gnd/src/main/java/com/google/android/gnd/repository/OfflineAreaRepository.java

```diff
@@ -38,42 +41,63 @@
 import io.reactivex.Single;
 import java.io.File;
 import java.io.IOException;
+import java.net.URL;
 import javax.inject.Inject;
+import org.apache.commons.io.FileUtils;
 import timber.log.Timber;
 
 public class OfflineAreaRepository {
   private final TileDownloadWorkManager tileDownloadWorkManager;
   private final LocalDataStore localDataStore;
+  private final ProjectRepository projectRepository;
   private final GeoJsonParser geoJsonParser;
   private final FileUtil fileUtil;
+  private final Schedulers schedulers;
 
   private final OfflineUuidGenerator uuidGenerator;
 
   @Inject
   public OfflineAreaRepository(
       TileDownloadWorkManager tileDownloadWorkManager,
       LocalDataStore localDataStore,
+      ProjectRepository projectRepository,
       GeoJsonParser geoJsonParser,
       OfflineUuidGenerator uuidGenerator,
-      FileUtil fileUtil) {
+      FileUtil fileUtil,
+      Schedulers schedulers) {
     this.tileDownloadWorkManager = tileDownloadWorkManager;
     this.localDataStore = localDataStore;
     this.geoJsonParser = geoJsonParser;
     this.uuidGenerator = uuidGenerator;
+    this.projectRepository = projectRepository;
     this.fileUtil = fileUtil;
+    this.schedulers = schedulers;
   }
 
-  private Completable enqueueTileDownloads(OfflineArea area) {
-    File jsonSource;
-
-    try {
-      jsonSource = fileUtil.getFileFromRawResource(R.raw.gnd_geojson, Config.GEO_JSON);
-    } catch (IOException e) {
-      return Completable.error(e);
+  /**
+   * Download the offline basemap source for the active project.
+   *
+   * <p>Only the first basemap source is used. Sources are always re-downloaded and overwritten on
+   * subsequent calls.
+   */
+  private File downloadOfflineBaseMapSource(
+      ImmutableList<OfflineBaseMapSource> offlineBaseMapSources)
+      throws IOException, NoBaseMapSourceException {
+    if (offlineBaseMapSources.isEmpty()) {
+      throw new NoBaseMapSourceException("No basemap sources specified for this project.");
     }
 
-    ImmutableList<Tile> tiles = geoJsonParser.intersectingTiles(area.getBounds(), jsonSource);
+    OfflineBaseMapSource source = offlineBaseMapSources.get(0);
```

Comment:
How about passing only the first `OfflineBaseMapSource` to this method?
Since the method is named `downloadOfflineBaseMapSource`, the error "No basemap sources specified for this project." doesn't seem very intuitive. So that check should be done before the call reaches this function.
Wdyt?

Reply:
80e58f8 thanks!